content/docs/command-reference/exp/run.md

```diff
@@ -85,9 +85,10 @@ process finishes or gets interrupted (e.g. with Ctrl + `C`), DVC will
 <abbr>workspace</abbr> (overwriting any further changes done by the stage).
 
 `dvc exp run` again will continue from this point (useful for interrupted runs).
-Use `--reset` to roll-back the workspace to `HEAD` and restart the whole
-experiment. Alternatively, you can use `--rev` to continue from a specific
-(previous) checkpoint.
+Use `--reset` to reset (remove) any existing `checkpoint` outputs in your
+workspace before running the experiment. Note that `--reset` overrides any
+existing `dvc.lock` entries for `checkpoint` outputs. Alternatively, you can use
```

Comment:
> `--reset` overrides any existing `dvc.lock` entries for `checkpoint` outputs

I'm not getting this note, what are we trying to warn about?

Reply:
If your `dvc.lock` file is git-committed and the HEAD version of `dvc.lock` has a hash for the specified `checkpoint` out, that hash will be ignored on `--reset`, and the out will be completely removed before reproducing the pipeline (rather than being checked out to HEAD's hash).

Reply:
To put this in user terms, if you save a model at each checkpoint so that its weights get loaded as the starting point for the next training epoch, `--reset` will drop the model so that training will start without any existing model weights.